overlord/snapstate/snapmgr.go

```diff
@@ -546,6 +546,8 @@ func Manager(st *state.State, runner *state.TaskRunner) (*SnapManager, error) {
 	runner.AddHandler("migrate-snap-home", m.doMigrateSnapHome, m.undoMigrateSnapHome)
 	// no undo for now since it's last task in valset auto-resolution change
 	runner.AddHandler("enforce-validation-sets", m.doEnforceValidationSets, nil)
+	runner.AddHandler("pre-download-snap", m.doPreDownloadSnap, nil)
+	runner.AddHandler("release-inhibit", m.doReleaseInhibitLock, nil)
```

Comment:
I would really prefer if we could work out how not to need this kind of task, also see my comment about the contradiction about being run no matter what and task wait logic